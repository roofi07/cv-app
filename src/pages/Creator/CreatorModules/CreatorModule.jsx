import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { SubHeader, TextRow, Modules } from './modules'
import './CreatorModule.scss'


export class CreatorModule extends React.Component {
  static propTypes = {
    moduleData: ImmutablePropTypes.map,
    onRemove: PropTypes.func,
    onUpdate: PropTypes.func,
  }

  constructor() {
    super()

    this.updateData = this.updateData.bind(this)
    this.remove = this.remove.bind(this)
  }

  getModule() {
    const { moduleData } = this.props

    switch (moduleData.get('type')) {
      case Modules.TEXT_ROW:
        return <TextRow data={moduleData.get('data')} onUpdate={this.updateData} />
      case Modules.SUB_HEADER:
        return <SubHeader data={moduleData.get('data')} onUpdate={this.updateData} />
      default:
        return null
    }
  }

  remove() {
    const { moduleData, onRemove } = this.props
    onRemove(moduleData.get('id'))
  }

  updateData(data) {
    const { moduleData, onUpdate } = this.props
    const newData = moduleData.set('data', data)
    onUpdate(newData)
  }

  render() {
    return (
      <div className="creator-module">
        {this.getModule()}
        <button className="creator-module__remove-btn" onClick={this.remove} title="Remove"><i className="fa fa-ban" /></button>
      </div>
    )
  }
}

export default CreatorModule
