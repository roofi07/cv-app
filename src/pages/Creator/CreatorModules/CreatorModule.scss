.creator-module {
  position: relative;
  border: 1px dotted transparent;

  &:hover {
    border-color: #dedede;
  }

  &__remove-btn {
    position: absolute;
    top: -18px;
    right: -18px;
    width: 30px;
    height: 30px;
    font-size: 16px;
    color: #e24d4d;
    border-radius: 50%;
    border: 0;
    background-color: #f7f7f7;
    cursor: pointer;
    outline: 0;

    &:hover {
      background-color: #eee;
    }

    .creator-module:not(:hover) & {
      opacity: 0;
    }
  }
}
