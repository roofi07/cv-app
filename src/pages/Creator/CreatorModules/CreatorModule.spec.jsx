import React from 'react'
import { shallow } from 'enzyme'
import Immutable from 'immutable'
import { CreatorModule } from './CreatorModule'

const data = {
  data: {
    key: 'Test',
    value: 'test',
  },
  id: 1,
  type: 'TextRow',
}

const initPage = overrides => {
  const mockMethods = {
    onRemove: jest.fn(),
    onUpdate: jest.fn(),
  }
  const wrapper = shallow(<CreatorModule
    moduleData={Immutable.fromJS(data)}
    {...mockMethods}
    {...overrides}
  />)
  return { mockMethods, wrapper }
}

describe('CreatorModule', () => {
  it('should render as expected', () => {
    const { wrapper } = initPage()
    expect(wrapper).toMatchSnapshot()
  })
})
