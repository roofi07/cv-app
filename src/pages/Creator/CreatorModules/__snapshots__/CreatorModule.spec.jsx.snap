// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorModule should render as expected 1`] = `
<div
  className="creator-module"
>
  <TextRow
    data={
      Immutable.Map {
        key: "Test",
        value: "test",
      }
    }
    onUpdate={[Function]}
  />
  <button
    className="creator-module__remove-btn"
    onClick={[Function]}
    title="Remove"
  >
    <i
      className="fa fa-ban"
    />
  </button>
</div>
`;
